test/unittests/iotools/converters/adni_to_bids/adni_modalities/test_adni_fmap.py

```diff
@@ -0,0 +1,306 @@
+        (BIDSFMAPCase.TWO_PHASES_TWO_MAGNITUDES, "e2_ph", "phase2"),
+    ],
+)
+def test_phase_magnitude_renamer_success(case, input_value, expected):
+    from clinica.iotools.converters.adni_to_bids.adni_modalities.adni_fmap import (
+        phase_magnitude_renamer,
+    )
+
+    assert phase_magnitude_renamer(input_value, case) == expected
+
+
+@pytest.mark.parametrize(
+    "case, input_value",
+    [
+        (BIDSFMAPCase.ONE_PHASE_TWO_MAGNITUDES, "_e2"),
+        (BIDSFMAPCase.ONE_PHASE_TWO_MAGNITUDES, "e7"),
+        (BIDSFMAPCase.ONE_PHASE_TWO_MAGNITUDES, "ee1"),
+        (BIDSFMAPCase.TWO_PHASES_TWO_MAGNITUDES, "e2ph"),
+        (BIDSFMAPCase.TWO_PHASES_TWO_MAGNITUDES, " e2"),
+        (BIDSFMAPCase.TWO_PHASES_TWO_MAGNITUDES, "ea"),
+        (BIDSFMAPCase.TWO_PHASES_TWO_MAGNITUDES, "21"),
+    ],
+)
+def test_phase_magnitude_renamer_value_error(case, input_value):
+    from clinica.iotools.converters.adni_to_bids.adni_modalities.adni_fmap import (
+        phase_magnitude_renamer,
+    )
+
+    with pytest.raises(
+        ValueError, match=f"Extension {input_value} not taken in charge."
+    ):
+        phase_magnitude_renamer(input_value, case)
+
+
+@pytest.mark.parametrize(
+    "case, input_value",
+    [
+        (BIDSFMAPCase.DIRECT_FIELDMAPS, "e2"),
+        (BIDSFMAPCase.NOT_SUPPORTED, "e1"),
+        (BIDSFMAPCase.EMPTY_FOLDER, "toto"),
+    ],
+)
+def test_phase_magnitude_renamer_implemented_error(case, input_value):
+    from clinica.iotools.converters.adni_to_bids.adni_modalities.adni_fmap import (
+        phase_magnitude_renamer,
+    )
+
+    with pytest.raises(
+        NotImplementedError,
+        match=f"No renaming should be performed for case {case.value}",
+    ):
+        phase_magnitude_renamer(input_value, case)
+
+
+def test_rename_files_success_case1(tmp_path):
+    from clinica.iotools.converters.adni_to_bids.adni_modalities.adni_fmap import (
+        rename_files,
+    )
+
+    fmap = tmp_path / "fmap"
+    fmap.mkdir()
+    filenames = [
+        "sub01-sesM0_fmap_e1.nii.gz",
+        "sub01-sesM0_fmap_e1.json",
+        "sub01-sesM0_fmap_e2.nii.gz",
+        "sub01-sesM0_fmap_e2.json",
+        "sub01-sesM0_fmap_e2_ph.nii.gz",
+        "sub01-sesM0_fmap_e2_ph.json",
+        ".toto.txt",
+    ]
+    for f in filenames:
+        (fmap / f).touch()
+
+    expected_result = {
+        fmap / "sub01-sesM0_magnitude1.nii.gz",
+        fmap / "sub01-sesM0_magnitude1.json",
+        fmap / "sub01-sesM0_magnitude2.nii.gz",
+        fmap / "sub01-sesM0_magnitude2.json",
+        fmap / "sub01-sesM0_phasediff.nii.gz",
+        fmap / "sub01-sesM0_phasediff.json",
+        fmap / ".toto.txt",
+    }
+
+    rename_files(fmap, BIDSFMAPCase.ONE_PHASE_TWO_MAGNITUDES)
+
+    assert set(fmap.iterdir()) == expected_result
+
+
+def test_rename_files_success_case2(tmp_path):
+    from clinica.iotools.converters.adni_to_bids.adni_modalities.adni_fmap import (
+        rename_files,
+    )
```

Comment:
I believe you can simplify a bit these two tests since they share a lot of logic. I think something like the following would work (but I haven't tested it so you might need to tweak it a bit...):

```python
@pytest.fixture
def fmap_case_builder(tmp_path: Path, case: BIDSFMAPCase):
    fmap = tmp_path / "fmap"
    fmap.mkdir()
    filenames = [
        "sub01_sesM0_fmap_e1.nii.gz",
        "sub01_sesM0_fmap_e1.json",
        "sub01_sesM0_fmap_e2.nii.gz",
        "sub01_sesM0_fmap_e2.json",
        "sub01_sesM0_fmap_e2_ph.nii.gz",
        "sub01_sesM0_fmap_e2_ph.json",
        ".toto.txt",
    ]
    if case == BIDSFMAPCase.TWO_PHASES_TWO_MAGNITUDES:
        filenames += [
            "sub01_sesM0_fmap_e1_ph.nii.gz",
            "sub01_sesM0_fmap_e1_ph.json",
        ]
    for f in filenames:
        (fmap / f).touch()

@pytest.fixture
def expected(tmp_path: Path, case: BIDSFMAPCase) -> Set[Path]:
    expected_result = {
        fmap / "sub01-sesM0_magnitude1.nii.gz",
        fmap / "sub01-sesM0_magnitude1.json",
        fmap / "sub01-sesM0_magnitude2.nii.gz",
        fmap / "sub01-sesM0_magnitude2.json",
        fmap / ".toto.txt",
    }
    if case == BIDSFMAPCase.ONE_PHASE_TWO_MAGNITUDES:
        expected = expected.union(
            {fmap / "sub01-sesM0_phasediff.nii.gz", fmap / "sub01-sesM0_phasediff.json"}
        )
    if case == BIDSFMAPCase.TWO_PHASES_TWO_MAGNITUDES:
        expected = expected.union(
            {
                fmap / "sub01_sesM0_phase2.nii.gz",
                fmap / "sub01_sesM0_phase2.json",
                fmap / "sub01_sesM0_phase1.nii.gz",
                fmap / "sub01_sesM0_phase1.json",
            }
        )
    return expected

@pytest.mark.parametrize(
    "case", [BIDSFMAPCase.ONE_PHASE_TWO_MAGNITUDES, BIDSFMAPCase.TWO_PHASES_TWO_MAGNITUDES]
)
def test_rename_files_success(tmp_path, case, fmap_case_builder, expected):
    from clinica.iotools.converters.adni_to_bids.adni_modalities.adni_fmap import rename_file

    rename_fmap(tmp_path / "fmap", case)

    assert set((tmp_path / "fmap").iterdir()) == expected
```